src/main/java/edu/kit/kastel/sdq/artemis4j/grading/Annotation.java

```diff
@@ -24,6 +26,17 @@ public final class Annotation {
     private final AnnotationSource source;
     private String customMessage;
     private Double customScore;
+    // If not empty, this list contains classifiers that are used to group annotations.
+    // For example, all annotations that are related, could have the classifier ["a"],
+    // then they would be grouped together.
+    //
+    // You can add further classifiers to group annotations in a more fine-grained way:
+    // For example, when you have annotations with the classifiers ["a", "b"]
+    // and ["a", "c"], then if there are more than "annotationLimit"
+    // with the classifier "a", it would merge all annotations with the classifiers ["a", "b"]
+    // and all annotations with the classifiers ["a", "c"].
+    private final List<String> classifiers;
+    private final Optional<Integer> annotationLimit;
```

Comment:
Optional should not be a field. They should exclusively be used for return values :D